### PR TITLE
fix: CancelSendToEthereum refund coins to a different denom

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- convert the contract address to lower case in gravity denom

### CODE REVIEW

File: module/x/gravity/types/ethereum.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func NewSDKIntERC20Token(amount sdk.Int, contract common.Address) ERC20Token {
 
 // GravityCoin returns the gravity representation of the ERC20
 func (e ERC20Token) GravityCoin() sdk.Coin {
-	return sdk.Coin{Amount: e.Amount, Denom: strings.Join([]string{GravityDenomPrefix, e.Contract}, GravityDenomSeparator)}
+	return sdk.Coin{Amount: e.Amount, Denom: strings.Join([]string{GravityDenomPrefix, strings.ToLower(e.Contract)}, GravityDenomSeparator)}
```
Comment (jackzampolin) [inline module/x/gravity/types/ethereum.go:66]: This seems like a generally good idea and should help with weird issues around capitalization of eth addresses.

Comment (EricBolten) [reply module/x/gravity/types/ethereum.go:66]: Is there any concern that fields from `GravityCoin` end up in state anywhere that could cause a divergence based on the capitalization?

Comment (yihuang) [reply module/x/gravity/types/ethereum.go:66]: > Is there any concern that fields from `GravityCoin` end up in state anywhere that could cause a divergence based on the capitalization?

the unit test I added fails without this change, the `Denom` field is used to create coins in the bank module.

**Comment by zmanian** [comment]
My assessment is that this PR can be merged safely. 

it can cause a state divergence in `cancelSendToEthereum` so it's state breaking.

One question has been if we want to standardize on on lowercase representations or the checkSumHex representation. Lowercase is much faster to generate and that might be useful to the Cronus use case.

**Comment by EricBolten** [comment]
Hi @yihuang, closing this PR as this was a larger problem and was addressed here:

https://github.com/PeggyJV/gravity-bridge/pull/377

**Pull request closed without merging**